addons/binding/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/handler/MiIoAbstractHandler.java

```diff
@@ -0,0 +1,440 @@
+     */
+    protected void initalizeNetworkCache() {
+        network = new ExpiringCache<String>(CACHE_EXPIRY * 120, () -> {
+            try {
+                int ret = sendCommand(MiIoCommand.MIIO_INFO);
+                if (ret != 0) {
+                    return "id:" + ret;
+                }
+            } catch (Exception e) {
+                logger.debug("Error during network status refresh: {}", e.getMessage(), e);
+            }
+            return null;
+        });
+    }
+
+    protected void defineDeviceType(JsonObject miioInfo) {
+        updateProperties(miioInfo);
+        isIdentified = updateThingType(miioInfo);
+    }
+
+    private void updateProperties(JsonObject miioInfo) {
+        Map<String, String> properties = editProperties();
+        properties.put(Thing.PROPERTY_MODEL_ID, miioInfo.get("model").getAsString());
+        properties.put(Thing.PROPERTY_FIRMWARE_VERSION, miioInfo.get("fw_ver").getAsString());
+        properties.put(Thing.PROPERTY_HARDWARE_VERSION, miioInfo.get("hw_ver").getAsString());
+        if (miioInfo.get("wifi_fw_ver") != null) {
+            properties.put("wifiFirmware", miioInfo.get("wifi_fw_ver").getAsString());
+        }
+        if (miioInfo.get("mcu_fw_ver") != null) {
+            properties.put("mcuFirmware", miioInfo.get("mcu_fw_ver").getAsString());
+        }
+        updateProperties(properties);
+    }
+
+    protected boolean updateThingType(JsonObject miioInfo) {
+        String model = miioInfo.get("model").getAsString();
+        miDevice = MiIoDevices.getType(model);
+        if (configuration.model == null || configuration.model.isEmpty()) {
+            Configuration config = editConfiguration();
+            config.put(PROPERTY_MODEL, model);
+            updateConfiguration(config);
+            configuration = getConfigAs(MiIoBindingConfiguration.class);
+        }
+        if (!configuration.model.equals(model)) {
+            logger.info("Mi IO Device model {} has model config: {}. Unexpected unless manual override", model,
+                    configuration.model);
+        }
+        if (miDevice.getThingType().equals(getThing().getThingTypeUID())) {
+            logger.info("Mi IO model {} identified as: {}. Matches thingtype {}", model, miDevice.toString(),
+                    miDevice.getThingType().toString());
+            return true;
+        } else {
+            if (getThing().getThingTypeUID().equals(THING_TYPE_MIIO)) {
+                changeType(model);
+            } else {
+                logger.warn(
+                        "Mi IO Device model {} identified as: {}, thingtype {}. Does not matches thingtype {}. Unexpected, unless unless manual override.",
+                        miDevice.toString(), miDevice.getThingType(), getThing().getThingTypeUID().toString(),
+                        miDevice.getThingType().toString());
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @param model
```

Comment:
please fix or remove the rather empty JavaDoc completely

Reply:
@marcelv To see good your javadoc is you can run `mvn javadoc:javadoc` in the root of your binding. This will generate the javadoc documentation, But more interesting here it also prints a list of javadoc issues on the command line during generation.

Reply:
thanks for this tip.. cleaned all errors & warnings now